src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java

```diff
@@ -67,22 +80,50 @@ public AbstractMavenArtifactChoiceListProvider(final String artifactId) {
 
     @Override
     public List<String> getChoiceList() {
-
+    	StaplerRequest req = Stapler.getCurrentRequest();
+    	Item item = req.findAncestorObject(Item.class);
+    	
+    	IVersionReader serviceInstance = createServiceInstance(item);
+    	
         LOGGER.log(Level.FINE, "retrieve the versions from the repository");
-        final Map<String, String> mChoices = readURL(createServiceInstance(), getRepositoryId(), getGroupId(), getArtifactId(), getPackaging(), getClassifier(),
+        final Map<String, String> mChoices = readURL(serviceInstance, getRepositoryId(), getGroupId(), getArtifactId(), getPackaging(), getClassifier(),
                 getInverseFilter(), getFilterExpression(), getReverseOrder());
         
         // FIXME: CHANGE-1: Return only the keys, that are shorter then the values
         // return new ArrayList<String>(mChoices.keySet());
         return new LinkedList<String>(mChoices.values());
     }
 
+    @Extension
+    public static class DescriptorImpl extends Descriptor<ChoiceListProvider> {
+
+		public ListBoxModel doFillCredentialsIdItems(Item item, String credentialsId) {
```

Comment:
Does this work without annotations?
```suggestion
		public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item item, @QueryParameter String credentialsId) {
```